domain/servicefactory/model.go

```diff
@@ -5,26 +5,47 @@ package servicefactory
 
 import (
 	"github.com/juju/juju/core/changestream"
+	"github.com/juju/juju/domain/model"
+	modelservice "github.com/juju/juju/domain/model/service"
 )
 
 // ModelFactory provides access to the services required by the apiserver.
 type ModelFactory struct {
-	modelDB changestream.WatchableDBFactory
-	logger  Logger
+	logger   Logger
+	modelDB  changestream.WatchableDBFactory
+	provider ModelServiceProvider
+	uuid     model.UUID
+}
+
+// ModelServiceProvider is responsible for providing a bridge to the controller
+// database where the model service operations on.
+type ModelServiceProvider interface {
+	// ModelServiceForUUID returns a model service for the provided model UUID.
+	ModelServiceForUUID(model.UUID) *modelservice.Service
```

Comment:
We've currently not tied a db/domain/service to an actual uuid, we've currently just talked about a namespace (essentially just a string).

But at this junction we __do__ actually want to talk about the concrete value: model uuid. If we want something different we can add more specific method.

(I've convinced myself this is correct)

Reply:
Let me speak to @manadart about this.

Reply:
I'm not sure that instituting this structural pattern pays for itself.

What do we get over just including the model UUID in the `SetCloudCredential` service call?

Reply:
As discussed, have decoupled the concept out. :)